### PR TITLE
firmware: log exceptions in debug mode (no silent unhandled exceptions)

### PR DESCRIPTION
**Pull request by malinah**
If the exceptions are not handled as part of the program logic at least log them. Now you can find out when you run out of memory etc.

### CODE REVIEW

File: src/boot.py [hunk 1]
```diff
@@ -18,7 +18,8 @@
                     return
                 else:
                     label = "Wrong PIN, enter again"
-        except:  # noqa: E722
+        except Exception as e:  # noqa: E722
```
Comment (prusnak) [inline src/boot.py]: remove `# noqa: E722`, not needed anymore

File: src/boot.py [hunk 2]
```diff
@@ -18,7 +18,8 @@
...
+            log.exception(__name__, e)
```
Comment (prusnak) [inline src/boot.py]: Maybe put this in `if __debug__:`

File: src/boot.py [hunk 3]
```diff
@@ -18,7 +18,8 @@
...
             pass
```
Comment (prusnak) [inline src/boot.py]: Remove `pass`, not needed anymore

**Comment by jpochyla** [comment]
❤️

**Pull request merged**